packages/utils-network/tests/as-path-string.ts

```diff
@@ -13,6 +13,22 @@ test('asPathString returns the path name of an "http://" URL', (t) => {
     t.is(actual, expected, `asPathString doesn't return the path name of an http:// URL`);
 });
 
+test('asPathString returns the encoded path name of an "https://" URL', (t) => {
+    const url = new URL('https://myresource.com/my/path/%5B-dont-%20-decode-%5D');
+    const expected = '/my/path/%5B-dont-%20-decode-%5D';
+    const actual = asPathString(url);
+
+    t.is(actual, expected, `asPathString doesn't return the encoded path name of an https:// URL`);
+});
+
+test('asPathString returns the decoded path name of an "file:///" URL', (t) => {
+    const url = new URL('file:///my/path/%5Bdecode%20me%5D');
+    const expected = '/my/path/[decode me]';
```

Comment:
The expected value will vary by OS (which is why this test is currently failing on Windows). Probably should do something similar to the following test in this file.